client/src/app/shared/components/data-table/data-table.component.ts

```diff
@@ -66,12 +66,11 @@ export class DataTableComponent implements OnInit, OnDestroy {
   }
 
   isUrl(value: string) {
-    const pattern = new RegExp('^(https?:\\/\\/)?' + // protocol
-      '((([a-z\\d]([a-z\\d-]*[a-z\\d])*)\\.)+[a-z]{2,}|' + // domain name
-      '((\\d{1,3}\\.){3}\\d{1,3}))' + // OR ip (v4) address
-      '(\\:\\d+)?(\\/[-a-z\\d%_.~+]*)*' + // port and path
-      '(\\?[;&a-z\\d%_.~+=-]*)?' + // query string
-      '(\\#[-a-z\\d_]*)?$', 'i'); // fragment locator
-    return !!pattern.test(value);
+    try {
+      const url = new URL(value);
+    } catch (_) {
```

Comment:
this is a super neat way of checking for URLs! Tiny (personal choice) thing, I don't think you need the `(_)` at all, `catch { }` works fine and reads a bit better

Reply:
Also this function is useful in other places, maybe consider extracting it to a `shared.utils` module

Reply:
The problem of this code is that giving us lots of warnings like this: `sanitizing unsafe URL value OBI:0000415`

Reply:
But I could not find a working solution with sanitiser :-(

Reply:
Extracted